### ISSUE TITLE
Incorrect cross validation usage

### ISSUE THREAD

**Issue by RayceRossum** [UBC-MDS/DSCI_522_Pokemon_Legendariness_Classifier#21]
Jack Yang [11:46 AM]
Pasted image at 2018-11-24, 11:46 AM 

hmm was looking through our code
I think for cross validation, we don't need to split into train set and test set

Rayce Rossum [11:48 AM]
I think we do.. we always need a train and test

Jack Yang [11:49 AM]
at each depth we want to be passing the entire dataset, it splits for us and gives the overall test accuracy at that depth

Rayce Rossum [11:49 AM]
The entire training dataset though
Cause if we do that, then our test data is now influencing our training I believe
I remember asking about this
But I see what you're saying..
Let me do some research
As I understand it, overfitting is the result of model selection based on training and testing using the same data, where you have a flexible fitting mechanism: you fit your sample of data so closely that you're fitting the noise, outliers, and all the other variance.

Splitting the data into a training and testing set keeps you from doing this. But a static split is not using your data efficiently and your split itself could be an issue. Cross-validation keeps the don't-reward-an-exact-fit-to-training-data advantage of the training-testing split, while also using the data that you have as efficiently as possible (i.e. all of your data is used as training and testing data, just not in the same run).
Just tested and the result is about the same..
Smaller tree, similar accuracy

Jack Yang [11:57 AM]
oh?

Rayce Rossum [11:57 AM]
Should we update to use these results?
We always get depth 3 now, with those features being the best indicators

Jack Yang [11:58 AM]
yeah right cause we aren't splitting the dataset anymore
before I think we are running cross val on 25% of the dataset
so there may be fluctuations
also for the final fit, we should use the entire dataset

Rayce Rossum [11:59 AM]
Yea, that makes sense
Kk, i'll push that change
